### ISSUE TITLE
Fix for a bug I introduced in an old pull request

### ISSUE THREAD

**Issue by dtegunov** [3dem/relion#710]
A pull request I submitted a long time ago to enable support for FFTW-formatted 3D CTFs contained a bug that leads to crashes on some systems. [ml_optimiser.cpp, line 8342](https://github.com/3dem/relion/blob/9f3bf18ea82d8532993f1580f479553a200f4ebe/src/ml_optimiser.cpp#L8342), where Fctf is resized, should be placed before the conditional block:

```c++
// If there is a redundant half, get rid of it
if (XSIZE(Ictf()) == YSIZE(Ictf()))  
{  
	// Set the CTF-image in Fctf  
	Ictf().setXmippOrigin();  
	Fctf.resize(current_image_size, current_image_size, current_image_size / 2 + 1);  
	FOR_ALL_ELEMENTS_IN_FFTW_TRANSFORM(Fctf)  
	{  
		// Use negative kp, ip and jp indices, because the origin in the ctf_img lies half a pixel to the right of the actual center....  
		DIRECT_A3D_ELEM(Fctf, k, i, j) = A3D_ELEM(Ictf(), -kp, -ip, -jp);  
	}  
}
// otherwise, just window the CTF to the current resolution
else if (XSIZE(Ictf()) == YSIZE(Ictf()) / 2 + 1)
{
	windowFourierTransform(Ictf(), Fctf, YSIZE(Fctf));
}
```

becomes

```c++
Fctf.resize(current_image_size, current_image_size, current_image_size / 2 + 1);

// If there is a redundant half, get rid of it
if (XSIZE(Ictf()) == YSIZE(Ictf()))
{
	// Set the CTF-image in Fctf
	Ictf().setXmippOrigin();
	FOR_ALL_ELEMENTS_IN_FFTW_TRANSFORM(Fctf)
	{
		// Use negative kp, ip and jp indices, because the origin in the ctf_img lies half a pixel to the right of the actual center....
		DIRECT_A3D_ELEM(Fctf, k, i, j) = A3D_ELEM(Ictf(), -kp, -ip, -jp);
	}
}
// otherwise, just window the CTF to the current resolution
else if (XSIZE(Ictf()) == YSIZE(Ictf()) / 2 + 1)
{
	windowFourierTransform(Ictf(), Fctf, YSIZE(Fctf));
}
```

Sorry for not catching it before. Let me know if you'd like to have this as a pull request instead.

**Comment by biochem-fan** [comment]
OK, fixed in `ca75bf9`.

@joton In 3.2 (`master` of `relion-devel`), this part of the code has already been refactored and does not seem to be affected by this.

**Comment by joton** [comment]
Hi,
I already fixed that when reimplemented the way CTF files are read for subtomo. Actually, that code has not been in master branch till yesterday when we merged it, that's the reason Takanori couldn't  see it.

Anyway, Thanks a lot @dtegunov and @biochem-fan!
